### PR TITLE
Use argparse for YAML generation. Allow optional strict checking

### PR DESCRIPTION
**Pull request by SebastianSchildt**
Using argparse for arguments parsing, removing the outdated getopt code.

Main advantage. Less code, and autogenerated usage

See also #105

### CODE REVIEW

**Review by erikbosch — APPROVED** [review]
Looks ok to me

**Comment by erikbosch** [comment]
Meeting: ready to merge

File: vspec2yaml.py [hunk 1]
```diff
@@ -92,36 +79,33 @@ def write_line_break(self, data=None):
...
+    parser.add_argument('-I', '--include-dir', action='append',  metavar='dir', type=str,
+                    help='Add include directory to search for included vspec files.')
+    parser.add_argument('-s', '--strict', action='store_true', help='Use strict checking: Terminate when non-core attribute is found.' )
+    parser.add_argument('--no-uuid', action='store_true', help='Exclude uuids from generated yaml.' )
+    parser.add_argument('vspec_file', metavar='<vspec_file>', help='The vehicle specification file to convert.')
+    parser.add_argument('yaml_file', metavar='<yaml file>', help='The file to output the YAML objects to.')
+
+    args = parser.parse_args()
+
+    strict=args.strict
```
Comment (gunnarx) [inline vspec2yaml.py:94]: Missing spaces around =

File: vspec2yaml.py [hunk 2]
```diff
@@ -92,36 +79,33 @@ def write_line_break(self, data=None):
...
+    generate_uuids=not args.no_uuid
```
Comment (gunnarx) [inline vspec2yaml.py:95]: ^

Comment (erikbosch) [reply vspec2yaml.py:95]: @SebastianSchildt -will you fix what Gunnar detected and update the PR?

**Comment by gunnarx** [comment]
I tried to fix and force push but boschglobal:feature/argparse_for_yaml is not accepting me to push.  
(This is a setting somewhere when you create the PR)

File: vspec2yaml.py (hunk 1 above)
Comment (SebastianSchildt) [reply vspec2yaml.py:94]: sorry, didn't see it, will fix

File: vspec2yaml.py (hunk 2 above)
Comment (SebastianSchildt) [reply vspec2yaml.py]: Fixed

**Pull request merged**